ray-operator/apis/utils/util.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import (
+	"fmt"
+	"math"
+	"unicode"
+)
+
+const (
+	ServeName = "serve"
+)
+
+// GenerateServeServiceName generates name for serve service.
+func GenerateServeServiceName(serviceName string) string {
+	return CheckName(fmt.Sprintf("%s-%s-%s", serviceName, ServeName, "svc"))
+}
+
+// GenerateServeServiceLabel generates label value for serve service selector.
+func GenerateServeServiceLabel(serviceName string) string {
+	return fmt.Sprintf("%s-%s", serviceName, ServeName)
+}
+
+// CheckName makes sure the name does not start with a numeric value and the total length is < 63 char
+func CheckName(s string) string {
+	maxLength := 50 // 63 - (max(8,6) + 5 ) // 6 to 8 char are consumed at the end with "-head-" or -worker- + 5 generated.
+
+	if len(s) > maxLength {
+		// shorten the name
+		offset := int(math.Abs(float64(maxLength) - float64(len(s))))
+		fmt.Printf("pod name is too long: len = %v, we will shorten it by offset = %v", len(s), offset)
+		s = s[offset:]
+	}
+
+	// cannot start with a numeric value
+	if unicode.IsDigit(rune(s[0])) {
+		s = "r" + s[1:]
+	}
+
+	// cannot start with a punctuation
+	if unicode.IsPunct(rune(s[0])) {
+		fmt.Println(s)
+		s = "r" + s[1:]
+	}
+
+	return s
+}
```

Comment:
These functions are moved from `ray-operator/controllers/ray/utils/util.go` to avoid import cycles.